### PR TITLE
added comments to example sketch

### PR DESCRIPTION
**Pull request by codeoftherings**
I have added new comments, and improved existing comments, to the example sketches of the following libraries:
AP_Airspeed
AP_Baro
AP_Compass
AP_GPS

### CODE REVIEW

File: libraries/AP_Baro/examples/BARO_generic/BARO_generic.cpp [hunk 1]
```diff
@@ -62,7 +87,9 @@ void loop()
                             (double)barometer.get_altitude(),
                             (double)barometer.get_climb_rate(),
                             (unsigned)read_time);
-    } else {
+    } 
+    //if stipulated time has not passed between two distinct readings, delay the program for a millisecond
+    else {
```
Comment (khancyr) [inline libraries/AP_Baro/examples/BARO_generic/BARO_generic.cpp]: don't break the code style for comments

**Comment by khancyr** [comment]
Hello,

Thanks for contributions ! Nevertheless, you commit message aren't good ! They should start with the name of the libray : example : 
AP_Airspeed: add more comments to example

**Comment by codeoftherings** [comment]
@khancyr Understood. I'll keep that in mind. Thanks for the feedback.

File: libraries/AP_Baro/examples/BARO_generic/BARO_generic.cpp (hunk 1 above)
Comment (codeoftherings) [reply libraries/AP_Baro/examples/BARO_generic/BARO_generic.cpp]: I have made the changes you asked for. Please check

Comment (magicrub) [reply libraries/AP_Baro/examples/BARO_generic/BARO_generic.cpp]: still here

Comment (codeoftherings) [reply libraries/AP_Baro/examples/BARO_generic/BARO_generic.cpp]: So you want else on a seperate line? that was how it was in the original PR

File: libraries/AP_Airspeed/examples/Airspeed/Airspeed.cpp [hunk 2]
```diff
@@ -80,4 +92,4 @@ const struct AP_Param::GroupInfo        GCS_MAVLINK::var_info[] = {
 };
 GCS_Dummy _gcs;
 
-AP_HAL_MAIN();
+AP_HAL_MAIN();
```
Comment (khancyr) [inline libraries/AP_Airspeed/examples/Airspeed/Airspeed.cpp]: We need to keep the newline at the end of the file

File: libraries/AP_Baro/examples/BARO_generic/BARO_generic.cpp (hunk 1 above)
Comment (khancyr) [reply libraries/AP_Baro/examples/BARO_generic/BARO_generic.cpp]: no the else shouldn't be break , it should stay } else { and put the comment after or under

File: libraries/AP_GPS/examples/GPS_AUTO_test/GPS_AUTO_test.cpp [hunk 3]
```diff
@@ -99,4 +114,4 @@ void loop()
 }
 
 // Register above functions in HAL board level
-AP_HAL_MAIN();
+AP_HAL_MAIN();
```
Comment (khancyr) [inline libraries/AP_GPS/examples/GPS_AUTO_test/GPS_AUTO_test.cpp]: don't remove the newline at the end of the file

**Comment by codeoftherings** [comment]
@khancyr please check

**Comment by peterbarker** [comment]
Sorry for the delay on getting this merged.
I've:
 - rebased it to fix conflicts
 - fixed some whitespace issues
 - pushed over @khancyr 's work who did it 32 seconds faster than I did.

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**